src/typechecker/FStar.TypeChecker.TcTerm.fst

```diff
@@ -4335,6 +4330,18 @@ and tc_trivial_guard env t =
   Rel.force_trivial_guard env g;
   t,c
 
+and tc_attributes env attrs =
+  // we don't want attributes to depend on local names (otherwise,
+  // this would imply dependent attributes, substitutions within
+  // attributes...) hence we get rid of [gamma] (the local names)
+  // below.
+  let env = { env with gamma = []; gamma_sig = []; gamma_cache = BU.smap_create 100 } in
+  try List.map (fun attr -> fst (tc_trivial_guard env attr)) attrs
+  with Error (e, msg, rng, ctx) -> // adding small hint for the user
+    raise (Error ( Fatal_VariableNotFound
+                 , msg ^ ". Hint: local names are not allowed in attributes."
+                 , rng, ctx))
+
```

Comment:
You can use `Errors.with_ctx` for this.

Reply:
Nice, thanks!

Reply:
Actually, I just figured out I only need to insert a hint when hitting a `Fatal_VariableNotFound`, but not in any other case, thus I cannot use `with_ctx`, and need to pattern match correctly on the exception.

Otherwise, I'm wondering: should I add an error specifically for attributes referring to local names, instead of just concatenating some hint to the error message? Would be cleaner but maybe that's a bit overkill. What do you think @mtzguido?